src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            Contact contact = contactsAddrs.contacts[i];
+            contact.MailingStreet = 'Direct Street Edit\r\nSecond Line';
+            contact.MailingCity = 'Direct City Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        // verify Contact address
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals('Direct Street Edit\r\nSecond Line', contact.MailingStreet);
+            system.assertEquals('Direct Street Edit\r\nSecond Line', contact.MailingStreet);
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify the Address objects split street.
+        List<Address__c> ListAddr = [select Id, MailingStreet__c, MailingStreet2__c, Formula_MailingStreetAddress__c, Default_Address__c from Address__c];
+        for (Address__c addr : ListAddr) {
+            if (addr.Default_Address__c) {
+                system.assertEquals('Direct Street Edit', addr.MailingStreet__c);
+                system.assertEquals('Second Line', addr.MailingStreet2__c);
+                system.assertEquals('Direct Street Edit, Second Line', addr.Formula_MailingStreetAddress__c);
+            }
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Update N organizational Contacts with Mailing addresses 
+    verify:
+        N addresses created
+        Contact addresses updated
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateRecordsWithAddrInfo() {         
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+ 
+        // create Contacts without addresses
+        Integer contactCount = 3;
+        List<Contact> contacts = UTIL_UnitTestData_TEST.createMultipleTestContacts(contactCount);
+        insert contacts;
+
+        // update the Contacts' address
+        for (Contact contact : contacts) {
+            contact.MailingStreet = '123 45th';
+            contact.MailingCity = 'Seattle';
+        }
+        Test.startTest();
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contacts;
+        Test.stopTest();
+
+        // verify results
+        contacts = [select Id, Name, MailingStreet, MailingCity, MailingState, MailingPostalCode, MailingCountry, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.